File: exercise/Week2/UserMVP/UserController.cs
public class UserController
{
    private readonly UserService _service;

    public UserController(UserService service)
    {
        _service = service;
    }

    public async Task<UserDto?> GetUser(int userId)
    {
        return await _service.GetUserAsync(userId);
    }

    public async Task<List<PostDto>> GetUserPosts(int userId)
    {
        return await _service.GetUserPostsAsync(userId);
    }

    public async Task<UserProfileDto?> GetUserProfile(int userId)
    {
        return await _service.GetUserProfileAsync(userId);
    }
}

File: exercise/Week2/ts_api_flow/UserController.ts
import { UserService } from "./UserService";
import { UserDto } from "./UserDto";

export class UserController {
    private _service = new UserService();

    async getUserById(id: string): Promise<UserDto | undefined> {
        return await this._service.getUserById(id);
    }

    async getUsers(): Promise<UserDto[]> {
        return await this._service.getUsers();
    }
}